sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/FastestServerFetcher.kt

```diff
@@ -113,12 +107,7 @@ internal class FastestServerFetcher(
             }
         }
 
-        val lightWalletClient =
-            LightWalletClient.new(
-                context = context,
-                lightWalletEndpoint = endpoint,
-                singleRequestTimeout = FETCH_THRESHOLD
-            )
+        val lightWalletClient = kotlin.runCatching { walletClientFactory.create(endpoint) }.getOrNull() ?: return null
```

Comment:
This will run some of the measurements over Tor and some direct. Other than making the data a bit noisier, I think this is fine. The tight measurement loop means that the Tor connections are definitely linkable to the direct connection, but that's not really a problem for the measurement phase (vs tx fetching or submission where it would be).